.github/workflows/mvn-verify-check.yml

```
@@ -30,7 +30,44 @@ env:
     -Dmaven.scaladoc.skip
 
 jobs:
+  cache-dependencies:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        java: [ '8']
+    steps:
+      - uses: actions/checkout@v3 # refs/pull/:prNumber/merge
+      - uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: ${{ matrix.java }}
+      - run: |
+          set -x
+          max_retry=3; delay=30; i=1
+          while true; do
+            mvn help:evaluate -pl dist -Dexpression=included_buildvers \
+            -DforceStdout -PnoSnapshots -q | tr -d ',' | \
+            xargs -n 1 bash -c 'mvn initialize -pl sql-plugin-api -am -Dbuildver=$1' _ && break || {
+            if [[ $i -le $max_retry ]]; then
+              echo "mvn command failed. Retry $i/$max_retry."; ((i++)); sleep $delay; ((delay=delay*2))
+            else
+              echo "mvn command failed. Exit 1"; exit 1
+            fi
+          }
+          done 
+
+      - name: Get current date
+        run: |
+          echo "DATE=$(date +'%Y-%m-%d')" >> $GITHUB_ENV
+
+      - name: Cache local Maven repository
+        uses: actions/cache@v3
+        with:
+          path: ~/.m2/repository
```

should it be just .m2 for all metadata? Any reason not to save/restore everything. The current runs still seem to download many dependencies although the cache action claims to have restored 2G of cache.

```suggestion
          path: ~/.m2
```